c++/src/Statistics.hh

```diff
@@ -605,6 +676,43 @@ namespace orc {
     }
   };
 
+  class StripeStatisticsImpl: public StripeStatistics {
+  private:
+    std::unique_ptr<StatisticsImpl> columnStats;
+    std::vector<std::vector<std::unique_ptr<const ColumnStatistics> > > rowIndexStats;
```

Comment:
auto_ptr is not copy-assignable so we can't create a container of auto_ptr. This is the problem.